src/Command/ModelCommand.php

```diff
@@ -783,11 +790,18 @@ public function getValidation(Table $model, array $associations, Arguments $args
 
         $validate = [];
         $primaryKey = $schema->getPrimaryKey();
+        $foreignKeys = [];
+        if (isset($associations['belongsTo'])) {
+            foreach ($associations['belongsTo'] as $assoc) {
+                $foreignKeys[] = $assoc['foreignKey'];
+            }
+        }
         foreach ($fields as $fieldName) {
             // Skip primary key
             if (in_array($fieldName, $primaryKey, true)) {
                 continue;
             }
+            $this->isCurrentFieldForeignKey = in_array($fieldName, $foreignKeys, true);
```

Comment:
I don't love this as a property. Couldn't this be a parameter to `fieldValidation()`?  As a property it will sometimes be stale. Is there a reason you went with a property?